pkg/machines/components/storagePools/createStoragePoolDialog.jsx

```diff
@@ -0,0 +1,327 @@
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Alert, Button, Form, FormGroup, Grid, Modal } from 'patternfly-react';
+
+import FileAutoComplete from 'cockpit-components-file-autocomplete.jsx';
+import * as Select from 'cockpit-components-select.jsx';
+import { createStoragePool, getAllStoragePools } from '../../actions/provider-actions.es6';
+import cockpit from 'cockpit';
+
+const _ = cockpit.gettext;
+
+const LIBVIRT_SESSION_CONNECTION = 'session';
+const LIBVIRT_SYSTEM_CONNECTION = 'system';
+
+const StoragePoolNameRow = ({ onValueChanged, dialogValues }) => {
+    return (
+        <FormGroup controlId='name'>
+            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
+                {_("Name")}
+            </Grid.Col>
+            <Grid.Col sm={6}>
+                <input id='storage-pool-input-name'
+                       type='text'
+                       minLength={1}
+                       placeholder={_("Storage Pool Name")}
+                       value={dialogValues.name || ''}
+                       onChange={e => onValueChanged('name', e.target.value)}
+                       className='form-control' />
+            </Grid.Col>
+        </FormGroup>
+    );
+};
+
+const StoragePoolHostRow = ({ onValueChanged, dialogValues }) => {
+    if (['netfs'].includes(dialogValues.type))
+        return (
+            <FormGroup controlId='host'>
+                <Grid.Col componentClass={Form.ControlLabel} sm={3}>
+                    {_("Host")}
+                </Grid.Col>
+                <Grid.Col sm={6}>
+                    <input id='storage-pool-input-host'
+                           type='text'
+                           minLength={1}
+                           placeholder={_("Host Name")}
+                           value={dialogValues.host || ''}
+                           onChange={e => onValueChanged('host', e.target.value)}
+                           className='form-control' />
+                </Grid.Col>
+            </FormGroup>
+        );
+    return null;
+};
+
+const StoragePoolTypeRow = ({ onValueChanged, dialogValues }) => {
+    const poolTypes = [
+        { type: 'dir', detail: 'Filesystem Directory' },
+        { type: 'netfs', detail: 'Network Exported Directory' },
```

Comment:
Is this the same as NFS or something different?

Reply:
Libvirt under nfs category has some other types like GFS. Anyway, I think we are interested only in nfs in cockpit, so I can rename the netfs to nfs.